File: sdks/ruby/lib/sdk_tpl.rb
# frozen_string_literal: true

require_relative "sdk_tpl/version"
require_relative "sdk_tpl/config"
require_relative "sdk_tpl/client"
require_relative "sdk_tpl/errors/error_codes"
require_relative "sdk_tpl/errors/sdk_tpl_error"
require_relative "sdk_tpl/errors/error_sanitizer"
require_relative "sdk_tpl/http/circuit_breaker"
require_relative "sdk_tpl/http/retry_handler"
require_relative "sdk_tpl/http/http_client"
require_relative "sdk_tpl/security/security"
require_relative "sdk_tpl/models/email_provider"
require_relative "sdk_tpl/models/email"
require_relative "sdk_tpl/validators/email_validators"

# SdkTpl Ruby SDK
#
# Provides a high-level client for interacting with the SdkTpl API,
# with built-in retry logic, circuit breaking, request signing, and error
# sanitization.
#
# @example Basic usage
#   client = SdkTpl::Client.new(api_key: "your-api-key")
#   health = client.health_check
#   puts health["status"]
#   client.close
module SdkTpl
end

File: sdks/dart/lib/sdk_tpl.dart
/// SdkTpl - Official Dart SDK.
///
/// Provides a type-safe client for interacting with the SdkTpl API.
///
/// ```dart
/// import 'package:sdk_tpl/sdk_tpl.dart';
///
/// final client = SdkTplClient(
///   SdkTplConfig(apiKey: 'your-api-key'),
/// );
///
/// final health = await client.healthCheck();
/// print(health.status);
/// ```
library sdk_tpl;

export 'src/client.dart';
export 'src/config.dart';
export 'src/errors/error_code.dart';
export 'src/errors/sdk_tpl_error.dart';
export 'src/errors/error_sanitizer.dart';
export 'src/http/http_client.dart';
export 'src/http/circuit_breaker.dart';
export 'src/http/retry_handler.dart';
export 'src/models/email_provider.dart';
export 'src/models/email_models.dart';
export 'src/security/security.dart';
export 'src/utils/version.dart';
export 'src/utils/logger.dart';
